test/io/print_layout.cpp

```diff
@@ -213,98 +214,227 @@ TEST_CASE("Print empty charge layout", "[print-charge-layout]")
+    hex_gate_lyt layout{aspect_ratio<hex_gate_lyt>{0, 0}};
 
     layout.create_or({}, {}, {0, 0});
 
     const charge_distribution_surface<sidb_cell_clk_lyt_siqad> cl{
         convert_to_siqad_coordinates(apply_gate_library<sidb_cell_clk_lyt, sidb_bestagon_library>(layout)),
         sidb_simulation_parameters{3, -0.32}, sidb_charge_state::NEGATIVE};
 
-    cl.assign_charge_state({46, 3, 0}, sidb_charge_state::NEUTRAL);
-    cl.assign_charge_state({72, 3, 0}, sidb_charge_state::NEGATIVE);
-    cl.assign_charge_state({48, 4, 0}, sidb_charge_state::NONE);
-    cl.assign_charge_state({70, 4, 0}, sidb_charge_state::NEGATIVE);
-    cl.assign_charge_state({52, 5, 0}, sidb_charge_state::NEGATIVE);
-    cl.assign_charge_state({66, 5, 0}, sidb_charge_state::NEUTRAL);
-    cl.assign_charge_state({54, 6, 0}, sidb_charge_state::NEGATIVE);
-    cl.assign_charge_state({64, 6, 0}, sidb_charge_state::POSITIVE);
-    cl.assign_charge_state({57, 9, 0}, sidb_charge_state::NEGATIVE);
-    cl.assign_charge_state({57, 10, 1}, sidb_charge_state::NONE);
-    cl.assign_charge_state({62, 10, 0}, sidb_charge_state::POSITIVE);
-    cl.assign_charge_state({59, 11, 1}, sidb_charge_state::NEGATIVE);
-    cl.assign_charge_state({59, 14, 0}, sidb_charge_state::NEGATIVE);
-    cl.assign_charge_state({60, 15, 0}, sidb_charge_state::NEGATIVE);
-    cl.assign_charge_state({64, 16, 0}, sidb_charge_state::POSITIVE);
-    cl.assign_charge_state({66, 17, 0}, sidb_charge_state::NEUTRAL);
-    cl.assign_charge_state({70, 18, 0}, sidb_charge_state::NEGATIVE);
-    cl.assign_charge_state({72, 19, 0}, sidb_charge_state::NEGATIVE);
+    cl.assign_charge_state({16, 3, 0}, sidb_charge_state::NEUTRAL);
+    cl.assign_charge_state({42, 3, 0}, sidb_charge_state::NEGATIVE);
+    cl.assign_charge_state({18, 4, 0}, sidb_charge_state::NONE);
+    cl.assign_charge_state({40, 4, 0}, sidb_charge_state::NEGATIVE);
+    cl.assign_charge_state({22, 5, 0}, sidb_charge_state::NEGATIVE);
+    cl.assign_charge_state({36, 5, 0}, sidb_charge_state::NEUTRAL);
+    cl.assign_charge_state({24, 6, 0}, sidb_charge_state::NEGATIVE);
+    cl.assign_charge_state({34, 6, 0}, sidb_charge_state::POSITIVE);
+    cl.assign_charge_state({27, 9, 0}, sidb_charge_state::NEGATIVE);
+    cl.assign_charge_state({32, 10, 0}, sidb_charge_state::POSITIVE);
+    cl.assign_charge_state({27, 10, 1}, sidb_charge_state::NONE);
+    cl.assign_charge_state({29, 11, 1}, sidb_charge_state::NEGATIVE);
+    cl.assign_charge_state({29, 14, 0}, sidb_charge_state::NEGATIVE);
+    cl.assign_charge_state({30, 15, 0}, sidb_charge_state::NEGATIVE);
+    cl.assign_charge_state({34, 16, 0}, sidb_charge_state::POSITIVE);
+    cl.assign_charge_state({36, 17, 0}, sidb_charge_state::NEUTRAL);
+    cl.assign_charge_state({40, 18, 0}, sidb_charge_state::NEGATIVE);
+    cl.assign_charge_state({42, 19, 0}, sidb_charge_state::NEGATIVE);
+
+    SECTION("Cropped")
+    {
+        std::stringstream print_stream{};
```

Comment:
You should be able to declare the stream outside of the sections. This would prevent the duplicate.